src/main/java/org/opentripplanner/middleware/docs/PublicApiDocGenerator.java

```diff
@@ -0,0 +1,416 @@
+package org.opentripplanner.middleware.docs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.opentripplanner.middleware.OtpMiddlewareMain;
+import org.opentripplanner.middleware.utils.YamlUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsText;
+
+/**
+ * Class that generates the public-facing API documentation.
+ */
+public class PublicApiDocGenerator {
+    private static final Logger LOG = LoggerFactory.getLogger(PublicApiDocGenerator.class);
+    private static final HttpClient client = HttpClient.newBuilder().build();
+    private static final String API_TEMPLATE_FILE = "api-swagger-template.yaml";
```

Comment:
Let's call this `SWAGGER_SUPPLEMENTS_FILE`/`swagger-supplements.yaml`

Reply:
Addressed  in https://github.com/ibi-group/otp-middleware/pull/59/commits/4e3253e4b58640c8a42db4a24d02ac2c3319a3a7